airflow/utils/helpers.py

```diff
@@ -307,7 +308,7 @@ def on_terminate(p):
             return
         # If operation not permitted error is thrown due to run_as_user, use sudo to kill the process
         if err.errno == errno.EPERM:
-            os.system('sudo kill -' + str(sig) + ' ' + str(os.getpgid(pid)))
+            subprocess.call('sudo -i kill -' + str(sig) + ' ' + str(os.getpgid(pid)), shell=True)
```

Comment:
`check_call.` And the point of using subprocess was to _avoid_ using a shell (and we don't want a login shell). And I asked for `--non-interactive` to be added.

Aaaan one more thing: this is going to run as root. Is it possible to run it as the user we are impersonating instead?

Reply:
I tried running it as the user impersonated(sudo -u) - i got the same operation not permitted error. 
sudo -n is the option for non-interactive, but there is no long form for it - I checked man page
i will use check_call()